File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/TimeJson.java
package ua.softgroup.matrix.server.supervisor.producer.json;

import com.fasterxml.jackson.annotation.JsonView;

import java.time.LocalDate;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class TimeJson {

    @JsonView(JsonViewType.OUT.class)
    private Integer todayMinutes;

    @JsonView({ JsonViewType.OUT.class, JsonViewType.IN.class })
    private Integer totalMinutes;

    @JsonView(JsonViewType.IN.class)
    private String reason;

    @JsonView(JsonViewType.IN.class)
    private LocalDate date;

    public TimeJson(Integer todayMinutes, Integer totalMinutes) {
        this.todayMinutes = todayMinutes;
        this.totalMinutes = totalMinutes;
    }

    public Integer getTotalMinutes() {
        return totalMinutes;
    }

    public void setTotalMinutes(Integer totalMinutes) {
        this.totalMinutes = totalMinutes;
    }

    public Integer getTodayMinutes() {
        return todayMinutes;
    }

    public void setTodayMinutes(Integer todayMinutes) {
        this.todayMinutes = todayMinutes;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return "TimeJson{" +
                "todayMinutes=" + todayMinutes +
                ", totalMinutes=" + totalMinutes +
                ", reason='" + reason + '\'' +
                ", date=" + date +
                '}';
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/TimeJson.scala
package ua.softgroup.matrix.server.supervisor.producer.json

import java.time.LocalDate

import com.fasterxml.jackson.annotation.JsonView

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class TimeJson {

  @JsonView(Array(classOf[JsonViewType.OUT]))
  @BeanProperty
  var todayMinutes: Int = _

  @JsonView(Array(classOf[JsonViewType.OUT], classOf[JsonViewType.IN]))
  @BeanProperty
  var totalMinutes: Int = _

  @JsonView(Array(classOf[JsonViewType.IN]))
  @BeanProperty
  var reason: String = _

  @JsonView(Array(classOf[JsonViewType.IN]))
  @BeanProperty
  var date: LocalDate = _

  def this(todayMinutes: Integer, totalMinutes: Integer) {
    this()
    this.todayMinutes = todayMinutes
    this.totalMinutes = totalMinutes
  }

  override def toString = s"TimeJson(todayMinutes=$todayMinutes, totalMinutes=$totalMinutes, reason=$reason, date=$date)"

}
